pkg/policy/policy_controller.go

```diff
@@ -635,6 +691,54 @@ func getTrigger(rd kyverno.ResourceDescription) []*kyverno.ResourceSpec {
 	return specs
 }
 
+func generateTriggers(lister listerv1.NamespaceLister, rule kyverno.Rule, log logr.Logger) []*kyverno.ResourceSpec {
+	var specs []*kyverno.ResourceSpec
+
+	nslist, err := lister.List(labels.NewSelector())
+	if err != nil {
+		log.Error(err, "failed to list namespaces")
+	}
+
+	triggers := genTrigger(nslist, rule.MatchResources.ResourceDescription)
+	specs = append(specs, triggers...)
+
+	for _, any := range rule.MatchResources.Any {
+		triggers := genTrigger(nslist, any.ResourceDescription)
+		specs = append(specs, triggers...)
+	}
+
+	triggers = []*kyverno.ResourceSpec{}
+	for _, all := range rule.MatchResources.All {
+		triggers = genTrigger(nslist, all.ResourceDescription)
+	}
+	return specs
+}
+
+func genTrigger(nslist []*v1.Namespace, rd kyverno.ResourceDescription) []*kyverno.ResourceSpec {
```

Comment:
This `genTrigger` only generates `resourceSpec` if a rule matches Namespace, what if a rule matches other kinds?

Reply:
Currently there will be no triggers its not a namespace, as we discussed we only want to support the namespace right ?